src/Features/CSharp/Portable/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace Microsoft.CodeAnalysis.CSharp.ConvertSwitchStatementToExpression
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal sealed partial class ConvertSwitchStatementToExpressionDiagnosticAnalyzer : AbstractBuiltInCodeStyleDiagnosticAnalyzer
+    {
+        public ConvertSwitchStatementToExpressionDiagnosticAnalyzer()
+            : base(IDEDiagnosticIds.ConvertSwitchStatementToExpressionDiagnosticId,
+                new LocalizableResourceString(nameof(CSharpFeaturesResources.Convert_switch_statement_to_expression), CSharpFeaturesResources.ResourceManager, typeof(CSharpFeaturesResources)),
+                new LocalizableResourceString(nameof(CSharpFeaturesResources.Use_switch_expression), CSharpFeaturesResources.ResourceManager, typeof(CSharpFeaturesResources)))
+        {
+        }
+
+        protected override void InitializeWorker(AnalysisContext context)
+            => context.RegisterOperationAction(AnalyzeOperation, OperationKind.Switch);
+
+        private void AnalyzeOperation(OperationAnalysisContext context)
+        {
+            var switchOperation = (ISwitchOperation)context.Operation;
+            if (switchOperation.Syntax.ContainsDiagnostics)
+            {
+                return;
+            }
+
+            var analysisResult = Analyzer.Analyze(switchOperation);
+            if (analysisResult.IsFailure || analysisResult.IsNeutral)
```

Comment:
what's the purpose of 'neutral' if it just causes bailing out?  seems like there's only need for 'success/failure'.

Reply:
when results are combined neutral can be overridden, failure can't. the result might remain neutral at the end, e.g. when all cases are `break;` alone.

Reply:
but .Analyze is the static method you provided.  so why not have that method just return Success/Failure?  The analysisResult is otherwise unused...